state/protocol/protocol_state/epochs/fallback_statemachine.go

```diff
@@ -18,16 +25,39 @@ var _ StateMachine = (*FallbackStateMachine)(nil)
 
 // NewFallbackStateMachine constructs a state machine for epoch fallback, it automatically sets
 // InvalidEpochTransitionAttempted to true, thereby recording that we have entered epoch fallback mode.
-func NewFallbackStateMachine(view uint64, parentState *flow.RichProtocolStateEntry) *FallbackStateMachine {
+// No errors are expected during normal operations.
+func NewFallbackStateMachine(params protocol.GlobalParams, view uint64, parentState *flow.RichProtocolStateEntry) (*FallbackStateMachine, error) {
 	state := parentState.ProtocolStateEntry.Copy()
-	state.InvalidEpochTransitionAttempted = true
+	nextEpochCommitted := state.EpochPhase() == flow.EpochPhaseCommitted
+	// we are entering fallback mode, this logic needs to be executed only once
+	if !state.InvalidEpochTransitionAttempted {
+		// the next epoch has not been committed, but possibly setup, make sure it is cleared
+		if !nextEpochCommitted {
+			state.NextEpoch = nil
+		}
```

Comment:
My gut feeling is similar to Jordans:
* I think we need an explicit representation for "currently in EFM mode"
* Our APIs were written with the mind-set that data once queryable is essentially never changed. My feeling is that we should try to change our APIs such that information about the upcoming epoch that has not been committed to by the protocol is not queryable. 

   While I think we have a few cases, where we need this data (like starting the DKG, collector root block voting), but they are very very specialized. Instead of covering these use-cases with the same API as all the other stuff caring only about committed epochs, I feel either a specialized API or just listening to the events would be sufficient for the protocol components that need uncommitted epoch information. 
   
+1 on this being work for separate PRs.

Reply:
One possible candidate is this issue: https://github.com/onflow/flow-go/issues/5723

Reply:
Added a note to [Open Questions](https://www.notion.so/flowfoundation/Spork-less-Epoch-Fallback-Recovery-Design-II-Epoch-Extensions-e923ca84b4d848c5a17de4b18fb5afd5?pvs=4#3f7e04110d6c4964805199f19d740cb6) to come back to this later.